compiler/src/main/kotlin/edu/cornell/cs/apl/viaduct/codegeneration/BackendCodeGenerator.kt

```diff
@@ -230,6 +230,8 @@ private class BackendCodeGenerator(
         private var tempMap: MutableMap<Pair<Temporary, Protocol>, String> = mutableMapOf()
         private var varMap: MutableMap<ObjectVariable, String> = mutableMapOf()
 
+        private var sendQueue: ArrayDeque<String> = ArrayDeque()
```

Comment:
Would be better to name this `selfSends`. It's more explicit.

Also, it's good practice to hide the implementation type and only expose the interface that you need:
```kotlin
private var selfSends: Queue<String> = LinkedList()
```